kadas/app/external/qgis/app/qgslayertreeviewindicatorprovider.h

```diff
@@ -0,0 +1,103 @@
+/***************************************************************************
+  qgslayertreeviewindicatorprovider.h - QgsLayerTreeViewIndicatorProvider
+
+ ---------------------
+ begin                : 17.10.2018
+ copyright            : (C) 2018 by Alessandro Pasotti
+ email                : elpaso@itopen.it
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSLAYERTREEVIEWINDICATORPROVIDER_H
+#define QGSLAYERTREEVIEWINDICATORPROVIDER_H
+
+#include "qgis_gui.h"
+
+#include <QObject>
+#include <QSet>
+#include <memory>
+
+#include "qgslayertreeviewindicator.h"
+
+class QgsLayerTreeNode;
+class QgsLayerTreeView;
+class QgsMapLayer;
+
+
+/**
+ * The QgsLayerTreeViewIndicatorProvider class provides an interface for
+ * layer tree indicator providers.
+ *
+ * Subclasses must override:
+ *
+ * - iconName()
+ * - tooltipText()
+ * - acceptLayer() filter function to determine whether the indicator must be added for the layer
+ *
+ * Subclasses may override:
+ *
+ * - onIndicatorClicked() default implementation does nothing
+ * - connectSignals() default implementation connects layers to dataSourceChanged()
+ * - disconnectSignals() default implementation disconnects layers from dataSourceChanged()
+ */
+class GUI_EXPORT QgsLayerTreeViewIndicatorProvider : public QObject
```

Comment:
```suggestion
class QgsLayerTreeViewIndicatorProvider : public QObject
```

`GUI_EXPORT` can only be used for classes which are compiled by QGIS, not for classes that are compiled by kadas.

Reply:
Ouch i copied over this file after i modified it in qgis. I forced pushed the file back to be equal to qgis